src/app/reports-handler.ts

```diff
@@ -17,19 +18,30 @@ import { logger } from '../common/logger';
  * @param fileExtensions {Array} array of file ext
  * @return {Promise<void>}
  */
-const generateArchiveForDirectory = (source: string, destination: string, fileExtensions: string[]): Promise<void> => {
+const generateArchiveForDirectory = (source: string, destination: string, fileExtensions: string[], retrievedLogs: ILogs[]): Promise<void> => {
 
     return new Promise((resolve, reject) => {
         logger.info(`reports-handler: generating archive for directory ${source}`);
         const output = fs.createWriteStream(destination);
         const archive = archiver('zip', { zlib: { level: 9 } });
+        const filesForCleanup: string[] = [];
 
         output.on('close', () => {
+            for (const file of filesForCleanup) {
+                if (fs.existsSync( file )) {
+                    fs.unlinkSync(file);
+                }
+            }
             logger.info(`reports-handler: generated archive for directory ${source}`);
             return resolve();
         });
 
         archive.on('error', (err: Error) => {
+            for (const file of filesForCleanup) {
```

Comment:
If for some reason archive fails won't we lose the actual logfile?

Reply:
Yes, I think this is desired behavior? These log files are just kept in memory and we only want to include them in the archive, not have them stored in file.

Reply:
No, these log files are stored in userData path and if the archive fails we lose both the file and archive right?

Reply:
No, these logs were received in response to 'collect-logs' being sent. They are not stored anywhere, just kept in memory. The only reason I save them as temp files is to be able to add them to the archive - after archiving is done, the temp files needs to be removed whether we succeeded or not.

It would be different if we had opted for a general "receive logs as we go and store them with the electron logs", but that isn't what we want. Instead, when the user goes to the menu and selects "Show logs in..." at that point we must retrieve the current in-memory logs, and add them to the same archive as we add the electron logs to.